### PR TITLE
[timeline] Sort timeline events before summarizing

### PR DESCRIPTION
**Pull request by iskakaushik**

### CODE REVIEW

**Review by flar — APPROVED** [review]
LGTM, I'll be curious to see what impact this has on our benchmark dashboard.

File: packages/flutter_driver/lib/src/driver/timeline.dart [hunk 1]
```diff
@@ -124,11 +120,20 @@ class TimelineEvent {
...
-    .map<TimelineEvent>((Map<String, dynamic> eventJson) => TimelineEvent(eventJson))
-    .toList();
+  final List<TimelineEvent> timelineEvents =
+      Iterable.castFrom<dynamic, Map<String, dynamic>>(jsonEvents)
+          .map<TimelineEvent>(
+              (Map<String, dynamic> eventJson) => TimelineEvent(eventJson))
+          .toList();
+
+  timelineEvents.sort((TimelineEvent e1, TimelineEvent e2) {
+    return e1.timestampMicros.compareTo(e2.timestampMicros);
```
Comment (flar) [inline packages/flutter_driver/lib/src/driver/timeline.dart:135]: I just tried running this on my favorite benchmark and got a null exception here. Apparently not all events have timestamps and so you need to do something special to compare events with null timestamps (null should be sorted as less than everything else, maybe?)

**Review by flar — CHANGES_REQUESTED** [review]

**Comment by iskakaushik** [comment]
reverted at https://github.com/flutter/flutter/pull/55769, will make `nulls` behave well and re-land.

File: packages/flutter_driver/lib/src/driver/timeline.dart (hunk 1 above)
Comment (flar) [reply packages/flutter_driver/lib/src/driver/timeline.dart:135]: It looks like all of the events in the timeline I was processing that had null timestamps were the "thread_name" events at the beginning of the log.

**Comment by flar** [comment]
Sorry about the premature approval. :(

**Pull request merged**